rosbag2/src/rosbag2/writer.cpp

```diff
@@ -105,4 +148,21 @@ bool Writer::should_split_bagfile() const
     return storage_->get_bagfile_size() > max_bagfile_size_;
   }
 }
+
+void Writer::finalize_metadata()
+{
+  metadata_.bag_size = 0;
```

Comment:
should that maybe be done in the `open` function, when `metadata_` is being initialized?

Reply:
This is just me being overly cautious in the event that future commits touch `metadata_`. `bag_size` is already set to 0 in `open` with the line: 
`metadata_ = rosbag2_storage::BagMetadata{};`.

Reply:
But what would you do in case it's not zero? The same with a call to `.clear`. What if there is already data in it (wrongly presumably)?
I guess what I am aiming at is, that if we do it (here), there should be a check for whatever reason it's not 0 and bring up an error message that there might be data lost/ignored. Otherwise, I think the call doesn't bring much.

Reply:
Actually I have a better reason why I did it this way:

I think the code is a little more clear that `metadata_.bag_size` is the sum of the size of all files since it gets set to 0 before the accumulating for-loop.